extensions/firebase/message_handler.go

```diff
@@ -132,6 +133,25 @@ func (h *messageHandler) HandleMessages(ctx context.Context, msg interfaces.Kafk
 	h.sendPush(ctx, km.Message, msg.Topic)
 }
 
+func (h *messageHandler) createDedupContentFromPayload(km kafkaFCMMessage) string {
+	contentData := make(map[string]interface{})
+
+	if km.Data != nil {
+		contentData["data"] = km.Data
+	}
+
+	if km.Notification != nil {
+		contentData["notification"] = km.Notification
+	}
+
+	contentJSON, err := json.Marshal(contentData)
+	if err != nil {
+		h.logger.WithError(err).Error("Error marshalling content data for deduplication")
+		return "error-marshalling-content-data-for-deduplication"
```

Comment:
Is this a pattern that we're using in this code? what would this accomplish? Do we want to store this as a message in redis? If we got an error we don't want to set anything in redis, right?